lib/setup_functions.php

```diff
@@ -0,0 +1,965 @@
+        fail("osm2pgsql not found in '$osm2pgsql'");
+    }
+
+    if (!is_null(CONST_Osm2pgsql_Flatnode_File) && CONST_Osm2pgsql_Flatnode_File) {
+        $osm2pgsql .= ' --flat-nodes '.CONST_Osm2pgsql_Flatnode_File;
+    }
+    if (CONST_Tablespace_Osm2pgsql_Data)
+        $osm2pgsql .= ' --tablespace-slim-data '.CONST_Tablespace_Osm2pgsql_Data;
+    if (CONST_Tablespace_Osm2pgsql_Index)
+        $osm2pgsql .= ' --tablespace-slim-index '.CONST_Tablespace_Osm2pgsql_Index;
+    if (CONST_Tablespace_Place_Data)
+        $osm2pgsql .= ' --tablespace-main-data '.CONST_Tablespace_Place_Data;
+    if (CONST_Tablespace_Place_Index)
+        $osm2pgsql .= ' --tablespace-main-index '.CONST_Tablespace_Place_Index;
+    $osm2pgsql .= ' -lsc -O gazetteer --hstore --number-processes 1';
+    $osm2pgsql .= ' -C '.$iCacheMemory;
+    $osm2pgsql .= ' -P '.$aDSNInfo['port'];
+    if (isset($aDSNInfo['username']) && $aDSNInfo['username']) {
+        $osm2pgsql .= ' -U ' . $aDSNInfo['username'];
+    }
+    if (isset($aDSNInfo['hostspec']) && $aDSNInfo['hostspec']) {
+        $osm2pgsql .= ' -H ' . $aDSNInfo['hostspec'];
+    }
+
+    $aProcEnv = null;
+    if (isset($aDSNInfo['password']) && $aDSNInfo['password']) {
+        $aProcEnv = array_merge(array('PGPASSWORD' => $aDSNInfo['password']), $_ENV);
+    }
+
+    $osm2pgsql .= ' -d '.$aDSNInfo['database'].' '.$aCMDResult['osm-file'];
+    runWithEnv($osm2pgsql, $aProcEnv);
+
+    $oDB =& getDB();
+    if (!$aCMDResult['ignore-errors'] && !chksql($oDB->getRow('select * from place limit 1'))) {
+        fail('No Data');
+    }
+}
+
+function createFunctions($aCMDResult)
+{
+    info('Create Functions');
+
+    if (!checkModulePresence()) {
+        fail('error loading nominatim.so module');
+    }
+
+    createSqlFunctions($aCMDResult);
+}
+
+function createTables($aCMDResult)
+{
+    info('Create Tables');
+ 
+    $sTemplate = file_get_contents(CONST_BasePath.'/sql/tables.sql');
+    $sTemplate = str_replace('{www-user}', CONST_Database_Web_User, $sTemplate);
+    $sTemplate = replaceTablespace(
+        '{ts:address-data}',
+        CONST_Tablespace_Address_Data,
+        $sTemplate
+    );
+    $sTemplate = replaceTablespace(
+        '{ts:address-index}',
+        CONST_Tablespace_Address_Index,
+        $sTemplate
+    );
+    $sTemplate = replaceTablespace(
+        '{ts:search-data}',
+        CONST_Tablespace_Search_Data,
+        $sTemplate
+    );
+    $sTemplate = replaceTablespace(
+        '{ts:search-index}',
+        CONST_Tablespace_Search_Index,
+        $sTemplate
+    );
+    $sTemplate = replaceTablespace(
+        '{ts:aux-data}',
+        CONST_Tablespace_Aux_Data,
+        $sTemplate
+    );
+    $sTemplate = replaceTablespace(
+        '{ts:aux-index}',
+        CONST_Tablespace_Aux_Index,
+        $sTemplate
+    );
+    pgsqlRunScript($aCMDResult, $sTemplate, false);
+
+    // re-run the functions
+    info('Recreate Functions');
+    createSqlFunctions($aCMDResult);
```

Comment:
The rerunning should also be part of the control flow and this function should create tables only as promised.